cmake/KokkosConfig.cmake.in

```
@@ -29,44 +29,38 @@ if("launch_compiler" IN_LIST Kokkos_FIND_COMPONENTS AND "separable_compilation"
-)
-  #
-  # if CUDA was enabled, the compilation language was not set to CUDA, and separable compilation was not
-  # specified, then set the RULE_LAUNCH_COMPILE and RULE_LAUNCH_LINK globally and
-  # kokkos_launch_compiler will re-direct to the compiler used to compile CUDA code during installation.
-  # kokkos_launch_compiler will re-direct if ${CMAKE_CXX_COMPILER} and -DKOKKOS_DEPENDENCE is present,
-  # otherwise, the original command will be executed
-  #
+    if((@KOKKOS_ENABLE_CUDA@ AND NOT KOKKOS_CXX_COMPILER_IS_COMPATIBLE) OR "launch_compiler" IN_LIST
+                                                                           Kokkos_FIND_COMPONENTS
+    )
+      #
+      # if find_package(Kokkos COMPONENTS launch_compiler) then rely on the
+      # RULE_LAUNCH_COMPILE and RULE_LAUNCH_LINK to always redirect to the
+      # appropriate compiler for Kokkos
+      #
 
-  # run test to see if CMAKE_CXX_COMPILER=nvcc_wrapper
-  kokkos_compiler_is_nvcc(IS_NVCC ${CMAKE_CXX_COMPILER})
+      message(
+        STATUS
+          "kokkos_launch_compiler is enabled globally. C++ compiler commands with -DKOKKOS_DEPENDENCE will be redirected to the appropriate compiler for Kokkos"
+      )
+      kokkos_compilation(GLOBAL)
 
-  # if not nvcc_wrapper and Kokkos_LAUNCH_COMPILER was not set to OFF
-  if(NOT IS_NVCC AND (NOT DEFINED Kokkos_LAUNCH_COMPILER OR Kokkos_LAUNCH_COMPILER))
-    message(
-      STATUS
-        "kokkos_launch_compiler is enabled globally. C++ compiler commands with -DKOKKOS_DEPENDENCE will be redirected to the appropriate compiler for Kokkos"
-    )
-    kokkos_compilation(GLOBAL)
+    else()
+      if(NOT KOKKOS_CXX_COMPILER_IS_COMPATIBLE)
+        message(
+          WARNING
+            "Kokkos can not compile a simple test program with the given CXX compiler ${CMAKE_CXX_COMPILER}. This might be due to incompatible flags or ABI."
+        )
+      endif()
```

Maybe we should directly error out instead of warning ...

---

One option might be to check a variable to bypass the test and make the test error on failure -- that way developers with a known good configuration can disable the test to save on configure time.

---

Since right now we do nothing, I would rather leave it at a warning to not force users that already have a working setup to add an additional definition. But more voices are appreciated